test/test_linalg.py

```diff
@@ -141,6 +142,31 @@ def run_test_case(a, b):
         run_test_case(zero_strided, b)
         run_test_case(a, zero_strided)
 
+    @dtypes(*all_types_and_complex_and(torch.half, torch.bfloat16, torch.bool))
```

Comment:
I don't think this is a great idea as the deprecated function will disappear at some point.

We would want to fabricate a situation where the warning raised is internal to the test and could be relied upon to not change the test outcome without changes to the warning code. That might be difficult to do.

Reply:
I'm not 100% sure I understand what you mean, but my impression is that since any normal usage of `TORCH_WARN_DEPRECATED` (and for that matter any `TORCH_WARN*` call) is subject to change, it would be nice to have some more reliable way to check that `TORCH_WARN_DEPRECATED` does the right thing, independent of any other operation.

We could add some function `torch._C._warn_deprecated()`, and others for other warning types, bound in `torch/csrc/autograd/FunctionsManual.cpp`, which just calls `TORCH_WARN*`, and we check that the correct warning type is raised in Python

Reply:
> I don't think this is a great idea as the deprecated function will disappear at some point.

We actually do this in a couple places and I think that's fine.
As long as you leave a comment there saying explicitly that the particular function is just as an example and it can be swapped if the warning is ever removed.

Reply:
I've added `torch._C._warn_deprecation` and `torch._C._warn` with a unit test for them